sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigEndianLongCoder.java

```diff
@@ -43,17 +43,25 @@ public static BigEndianLongCoder of() {
   private BigEndianLongCoder() {}
 
   @Override
-  public void encode(Long value, OutputStream outStream)
-      throws IOException, CoderException {
+  public void encode(Long value, OutputStream outStream) throws IOException, CoderException {
     if (value == null) {
       throw new CoderException("cannot encode a null Long");
     }
-    new DataOutputStream(outStream).writeLong(value);
+    encodeLong(value, outStream);
   }
 
   @Override
-  public Long decode(InputStream inStream)
-      throws IOException, CoderException {
+  public Long decode(InputStream inStream) throws IOException, CoderException {
+    return decodeLong(inStream);
+  }
+
+  /** A primitive version of {@link #encode(Long, OutputStream)}. */
+  public void encodeLong(long value, OutputStream outStream) throws IOException, CoderException {
```

Comment:
Instead of making this and decodeLong public, why not just move this code into InstantCoder and stop coupling the two coders together?